pkg/service/model_build_managed_sg_test.go

```diff
@@ -0,0 +1,243 @@
+				prefixListsConfigured: true,
+			},
+			wantErr: false,
+			want:    nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t1 *testing.T) {
+			annotationParser := annotations.NewSuffixAnnotationParser("service.beta.kubernetes.io")
+			task := &defaultModelBuildTask{
+				annotationParser: annotationParser,
+				service:          tt.fields.svc,
+			}
+			got, err := task.buildCIDRsFromSourceRanges(context.Background(), tt.fields.ipAddressType, tt.fields.prefixListsConfigured)
+			if tt.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_buildCIDRsFromSourceRanges_buildManagedSecurityGroupIngressPermissions(t *testing.T) {
+	type fields struct {
+		svc           *corev1.Service
+		ipAddressType elbv2model.IPAddressType
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    []ec2model.IPPermission
+		wantErr bool
+	}{
+		{
+			name: "default IPv4",
+			fields: fields{
+				svc: &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{},
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeNodePort,
+						Ports: []corev1.ServicePort{
+							{
+								Name:     "http",
+								Port:     80,
+								NodePort: 18080,
+							},
+						},
+					},
+				},
+				ipAddressType: elbv2model.IPAddressTypeIPV4,
+			},
+			wantErr: false,
+			want: []ec2model.IPPermission{
+				{
+					IPProtocol: "",
+					FromPort:   aws.Int64(80),
+					ToPort:     aws.Int64(80),
+					IPRanges: []ec2model.IPRange{
+						{
+							CIDRIP: "0.0.0.0/0",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "default dualstack",
+			fields: fields{
+				svc: &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"service.beta.kubernetes.io/aws-load-balancer-ip-address-type": "daulstack",
+						},
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeNodePort,
+						Ports: []corev1.ServicePort{
+							{
+								Name:     "http",
+								Port:     80,
+								NodePort: 18080,
+							},
+						},
+					},
+				},
+				ipAddressType: elbv2model.IPAddressTypeDualStack,
+			},
+			wantErr: false,
+			want: []ec2model.IPPermission{
+				{
+					IPProtocol: "",
+					FromPort:   aws.Int64(80),
+					ToPort:     aws.Int64(80),
+					IPRanges: []ec2model.IPRange{
+						{
+							CIDRIP: "0.0.0.0/0",
+						},
+					},
+				},
+				{
+					IPProtocol: "",
+					FromPort:   aws.Int64(80),
+					ToPort:     aws.Int64(80),
+					IPv6Range: []ec2model.IPv6Range{
+						{
+							CIDRIPv6: "::/0",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "no ip range but prefix list",
+			fields: fields{
+				svc: &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"service.beta.kubernetes.io/aws-load-balancer-security-group-prefix-lists": "pl-xxxxx",
```

Comment:
nit: can we test against multiple lists too?

Reply:
The test is added in [51c5998](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3584/commits/51c59981db512ff83c7c1297b7e7a597fac7d6ad)